plugins/MobileMessaging/Controller.php

```diff
@@ -155,4 +158,32 @@ public function getCredentialFields()
 
         return $view->render();
     }
+
+    public function getBrowserNotifications()
+    {
+        $login = null;
+        $token = Common::getRequestVar('token');
+
+        if ($token) {
+            $user = (new Model())->getUserByNotificationToken($token);
+            if ($user) {
+                $login = $user['login'];
+            }
+        }
+
+        if (! $login) {
+            throw new NoAccessException('Missing or invalid token');
+        }
+
+        $optionKey = 'ScheduledReports.notifications.' . $login;
+        $optionValue = Option::get($optionKey);
```

Comment:
Not sure I understand... any reason we are reading this from option instead of getting the actual notifications? Was it maybe needed because requesting the URL so often? When we request it only a few times, could avoid needing to use the options.

Reply:
I think I see why it's also done now. It could be too tricky to get all the user notifications from all sites and evaluate them, and knowing which ones have been delivered and which ones not etc.